### PR TITLE
Mkdocs for the documentation

### PR DESCRIPTION
**Pull request by susuhahnml**

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -21,27 +21,10 @@ def doc(session):
...
-        options = session.posargs[1:]
 
     session.install("-e", ".[doc]")
-    session.cd("doc")
-    if clean:
-        session.run("rm", "-rf", "_build")
-    session.run("sphinx-build", "-M", target, ".", "_build", *options)
-    if open_doc:
-        session.run("open", "_build/html/index.html")
+    session.run("open", "http://127.0.0.1:8000/systems/fillname/")
```
Comment (rkaminsk) [inline noxfile.py:26]: Can we handle this as before again? Maybe have a way to just build the documentation and one to serve it?
To build:
```
nox -s doc
```
To also serve:
```
nox -s doc serve
```
The docstring also has to be adjusted.

**Review by rkaminsk — CHANGES_REQUESTED** [review]
Just one small thing. I'll also give it a test run tomorrow.

**Comment by sthiele** [comment]
@susuhahnml Could we change the directory `/docs/reference/api` to `/docs/reference/API` to get the capitalization right?

**Comment by rkaminsk** [comment]
> @susuhahnml Could we change the directory `/docs/reference/api` to `/docs/reference/API` to get the capitalization right?

Please lets stick to all lower case.

**Comment by sthiele** [comment]
> Please lets stick to all lower case for filenames and urls.

Could one use the nav section in the mkdocs.yml file?

**Comment by rkaminsk** [comment]
I had a look regarding the navigation. It seems like mkdocs uses the filename to generate navitems and there is no way around except specifying the navigation in the mkdocs.yml or using plugins:

We could, for example, use <https://pypi.org/project/mkdocs-gen-nav-plugin/>.

**Comment by rkaminsk** [comment]
Another problem is that the yaml page preambles are currently rendered as markdown. In my tests, mkdocs seems to parse multimarkdown preambles. Maybe this is the default? We should decide what to use here.

**Comment by rkaminsk** [comment]
One last thing, the blue bar on top in dark mode looks kinda ugly.

**Comment by susuhahnml** [comment]
I was running away a bit from explicitly writing the navigation so that adding a file was enough and one less step. But I am also ok with keeping it as it gives us more flexibility. Regarding the color, we could use one of the default ones but I would also like to have something unique, since the default is used by many other systems. I also agree it does not look amazing in dark mode.

**Comment by susuhahnml** [comment]
I changed the color slightly, let me know if you like it better, I imagine that with the whole new branding we would ideally use whatever color is decided there. I also updated the readme with the -- serve for the docs

**Review by rkaminsk — APPROVED** [review]

**Pull request merged**